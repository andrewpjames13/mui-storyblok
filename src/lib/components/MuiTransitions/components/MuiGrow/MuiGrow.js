import React from 'react';
import PropTypes from 'prop-types';
import { Grow } from '@material-ui/core';
import useTransitionIn from '../../customHooks/useTransitionIn';
import StoryBlok from '../../../../utils/Storyblok';
import Blok from '../../../StoryBlokPage/components/Blok/Blok';
import { validComponentsRequired } from '../../../../utils/customProps';

const MuiGrow = ({
  rootClass,
  content,
  growIn,
  enter,
  exit,
}) => {
  const styles = StoryBlok.arrayToMuiStyles(rootClass);
  const transTime = useTransitionIn(growIn);

  let enterTime;
  if (enter === '0') {
    enterTime = 'auto';
  } else {
    enterTime = { enter: +enter, exit: +exit };
  }

  return (
    <div className={styles.root}>
      <Grow
        in={transTime}
        timeout={enterTime}
      >
        <div>
          {content.map((item, index) => (
            <Blok {...item} key={index} />
          ))}
        </div>
      </Grow>
    </div>
  );
};

export default MuiGrow;

MuiGrow.propTypes = {
  /**
   * stroyblok multiselect of css classes
   * Mui Override or extend the styles applied to the component.
   */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /**
   * mui props: number
   * Number of milliseconds for component to transition in.
   */
  growIn: PropTypes.string,
  /**
   * mui props: number
   * Duration in milliseconds to enter the screen.
   */
  enter: PropTypes.string,
  /**
   * mui props: number
   * Duration in milliseconds to leave the screen.
   */
  exit: PropTypes.string,
  /**
   * Content passed to render
   * components: Blok
   */
  content(props, propName, componentName) {
    return validComponentsRequired(props, propName, componentName, ['Blok'], 1);
  },
};

MuiGrow.defaultProps = {
  growIn: 300,
  enter: 1500,
  exit: 1500,
  rootClass: [],
  content: [],
};
