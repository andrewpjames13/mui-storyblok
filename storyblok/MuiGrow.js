module.exports = {
  "name": "MuiGrow",
  "display_name": null,
  "updated_at": "2020-05-15T20:45:19.327Z",
  "schema": {
    "rootClass": {
      "type": "options",
      "description": "* stroyblok multiselect of css classes\n* Mui Override or extend the styles applied to the component."
    },
    "content": {
      "type": "bloks",
      "restrict_components": true,
      "component_whitelist": [
        "Blok"
      ],
      "maximum": "1",
      "required": true
    },
    "growIn": {
      "type": "number",
      "description": "Number of milliseconds for component to transition in.",
      "default_value": "300"
    },
    "enter": {
      "type": "number",
      "description": "Duration in milliseconds to enter the screen.",
      "default_value": "1500"
    },
    "exit": {
      "type": "number",
      "description": "Duration in milliseconds to leave the screen.",
      "default_value": "1500"
    }
  },
  "image": null,
  "preview_field": null,
  "is_root": false,
  "preview_tmpl": null,
  "is_nestable": true,
  "all_presets": [],
  "preset_id": null,
  "real_name": "MuiGrow",
  "component_group_uuid": null
}