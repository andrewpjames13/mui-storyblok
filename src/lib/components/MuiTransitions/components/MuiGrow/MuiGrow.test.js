import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import MuiGrow from './MuiGrow';

function setup() {
  const props = {
    enter: 1500,
    exit: 1500,
    content: [{
      component: 'Blok',
      content: [{
        component: 'MuiGrid',
        content: [{
          component: 'MuiGridItem',
          content: [{
            component: 'MuiCard',
            content: [{
              component: 'MuiCardContent',
              content: [{
                component: 'MuiTypography',
                content: [{
                  component: 'MuiText',
                  text: 'Juice Box',
                }],
              }],
            }],
          }],
        }],
      }],
    }],
  };

  const comp = shallow(<MuiGrow {...props} />);
  return { comp, props };
}

describe('<MuiGrow />', () => {
  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(
      <MemoryRouter>
        <MuiGrow {...props} />
      </MemoryRouter>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render MuiGridList', () => {
    const { comp } = setup();
    expect(comp).toMatchSnapshot();
  });
});
