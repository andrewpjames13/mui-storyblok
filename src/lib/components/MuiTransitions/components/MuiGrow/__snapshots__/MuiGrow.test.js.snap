// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiGrow /> should render MuiGridList 1`] = `ShallowWrapper {}`;

exports[`<MuiGrow /> snapshot 1`] = `
<div
  className="makeStyles-root-2"
>
  <div
    style={
      Object {
        "opacity": 0,
        "transform": "scale(0.75, 0.5625)",
        "visibility": "hidden",
      }
    }
  />
</div>
`;
